src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,82 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TaskList;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.ui.Ui;
+
+/**
+ * Adds a task to the TaskList.
+ */
+public class AddCommand extends Command {
+    String type;
+    String details;
+
+    /**
+     * Constructs an AddCommand with the specified type and details.
+     *
+     * @param type The type of task for the command.
+     * @param details The details of the command.
+     */
+    public AddCommand(String type, String details) {
+        this.type = type;
+        this.details = details;
+    }
+
+    /**
+     * Adds the task to the TaskList and returns an acknowledgement message.
+     *
+     * @param tasks The TaskList where the task is to be added.
+     * @param ui The Ui that interacts with the user.
+     * @param storage The Storage to load and save tasks into the data file.
+     * @return A string with the message to be printed.
+     * @throws DukeException If the details is invalid.
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        Task task;
+        assert (type.equals("todo") || type.equals("deadline") || type.equals("event")) : "Invalid task type";
+        switch (type) {
+        case "todo":
+            if (details.trim().equals("")) {
```

Comment:
If this logic is repeated and common for all the tasks, then you extract it out of switch block